packages/@aws-cdk/aws-chatbot/lib/slack-channel-configuration.ts

```diff
@@ -0,0 +1,283 @@
+   *
+   * @default - A role will be created.
+   */
+  readonly configurationRole?: iam.IRole;
+
+  /**
+   * The ID of the Slack workspace authorized with AWS Chatbot.
+   *
+   * To get the workspace ID, you must perform the initial authorization flow with Slack in the AWS Chatbot console.
+   * Then you can copy and paste the workspace ID from the console.
+   * For more details, see steps 1-4 in Setting Up AWS Chatbot with Slack in the AWS Chatbot User Guide.
+   * @see https://docs.aws.amazon.com/chatbot/latest/adminguide/setting-up.html#Setup_intro
+   */
+  readonly slackWorkspaceId: string;
+
+  /**
+   * The ID of the Slack channel.
+   *
+   * To get the ID, open Slack, right click on the channel name in the left pane, then choose Copy Link.
+   * The channel ID is the 9-character string at the end of the URL. For example, ABCBBLZZZ.
+   */
+  readonly slackChannelId: string;
+
+  /**
+   * The SNS topics that deliver notifications to AWS Chatbot.
+   *
+   * @default None
+   */
+  readonly notificationTopics?: sns.ITopic[];
+
+  /**
+   * Specifies the logging level for this configuration.
+   * This property affects the log entries pushed to Amazon CloudWatch Logs.
+   *
+   * @default LoggingLevel.NONE
+   */
+  readonly loggingLevel?: LoggingLevel;
+}
+
+/**
+ * Logging levels include ERROR, INFO, or NONE.
+ */
+export enum LoggingLevel {
+  /**
+   * ERROR
+   */
+  ERROR = 'ERROR',
+
+  /**
+   * INFO
+   */
+  INFO = 'INFO',
+
+  /**
+   * NONE
+   */
+  NONE = 'NONE',
+}
+
+/**
+ * Represents a Slack channel configuration
+ */
+export interface ISlackChannelConfiguration extends cdk.IResource {
+
+  /**
+   * The ARN of the Slack channel configuration
+   * @attribute
+   */
+  readonly slackChannelConfigurationArn: string;
+
+  /**
+   * The name of Slack channel configuration
+   * @attribute
+   */
+  readonly configurationName: string;
+
+  /**
+   * The permission role of Slack channel configuration
+   * @attribute
+   *
+   * @default - A role will be created.
+   */
+  readonly configurationRole?: iam.IRole;
+}
+
+/**
+ * Either a new or imported Slack channel configuration
+ */
+abstract class SlackChannelConfigurationBase extends cdk.Resource implements ISlackChannelConfiguration {
+  abstract readonly slackChannelConfigurationArn: string;
+
+  abstract readonly configurationName: string;
+
+  abstract readonly configurationRole?: iam.IRole;
+
+  /**
+   * Adds extra permission to iam-role of Slack channel configuration
+   * @param statement
+   */
+  public addToPrincipalPolicy(statement: iam.PolicyStatement): void {
+    if (!this.configurationRole) {
+      return;
+    }
+
+    this.configurationRole!.addToPrincipalPolicy(statement);
+  }
+
+  /**
+   * Allows AWS chatbot to retrieve metric graphs from AWS CloudWatch.
+   */
+  public addNotificationPermissions(): void {
+    this.configurationRole!.addManagedPolicy(new iam.ManagedPolicy(this, 'NotificationsOnlyPolicy', {
```

Comment:
I think there's some inconsistency between how this method (and similar ones) below deal with role absence vs. `addToPrincipalPolicy`.  `addToPrincipalPolicy` does a noop if role doesn't exist... `addXPermissions` would throw.